pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -35,6 +35,20 @@ import (
 	crdv1b1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
 )
 
+var (
+	netlinkAdd         = netlink.LinkAdd
+	netlinkDel         = netlink.LinkDel
+	ensureRPF          = util.EnsureRPFilterOnInterface
+	netlinkSetUp       = netlink.LinkSetUp
+	netInterfaceByName = net.InterfaceByName
+	netlinkAddrAdd     = netlink.AddrAdd
+	netlinkAddrDel     = netlink.AddrDel
+	netlinkAddrList    = netlink.AddrList
```

Comment:
We have following 4 types supported in netlink interface, we can use mocked apis for these
netlink.AddrList
netlink.AddrDel
netlink.AddrAdd
netlink.LinkSetUp
However others like netlink.LinkAdd, netlink.LinkDel, net.InterfaceByName are not supported

Reply:
netlink.LinkAdd, netlink.LinkDel can be added to the interface

Reply:
This comment hasn't been resolved.

Reply:
Had Tried earlier replacing while holding the interface for assignee, it might not work unlike pkg/agent/route/route_linux.go as in route_linux.go, it is using netlink of type utilnetlink.Interface which matches type of interface pkg/agent/util/netlink/netlink_linux.go while replacing
```
// Client takes care of routing container packets in host network, coordinating ip route, ip rule, iptables and ipset.
type Client struct {
        netlink       utilnetlink.Interface
```
In assign function, it is directly using external netlink pkg

```
func (as *assignee) assign(ip net.IP, subnetInfo *crdv1b1.SubnetInfo) error {
        // If there is a real link, add the IP to its address list.
        if as.link != nil {
                addr := getIPNet(ip, subnetInfo)
                if err := netlink.AddrAdd(as.link, &netlink.Addr{IPNet: addr}); err != nil {
```
`
mockNetlink.EXPECT().AddrAdd(dummyDevice.Attrs().Name, &netlink.Addr{IPNet: expectedIPNet}).Return(nil)
`
For example above expectation are never met wrt interface defined in pkg/agent/util/netlink/netlink_linux.go ! similarly for all other netlink apis in ipassigner is used directly from netlink.